app/admin/events.rb

```diff
@@ -0,0 +1,46 @@
+ActiveAdmin.register Event do
+  menu label: 'All Events', priority: 3, parent: 'Administration'
+
+  actions :all, except: [:new, :edit, :update, :show, :destroy]
+
+  INDEX_COLUMNS_DEFINITION = proc do
+    column :id
+    column :eventable_type
+    column :eventable_id
+    column :event_type
+    column :title
+    column :description
+    column :date
+    column :url
+  end
+
+  index { instance_exec(&INDEX_COLUMNS_DEFINITION) }
+
+  csv { instance_exec(&INDEX_COLUMNS_DEFINITION) }
+
+  controller do
+    def scoped_collection
```

Comment:
to redirect to for example admin dashboard instead of displaying index page, you could do like

```
    def index
      super do |format|
        format.html { redirect_to admin_dashboard_path }
      end
    end
```
That would preserve format.csv response